### ISSUE TITLE
[Master] Async-Http-Client: Post a Chunked Body Test Failure

### ISSUE THREAD

**Issue by ChristopherDavenport** [http4s/http4s#1274]
Runs fine for me locally however we are getting consistent failure in Travis.

```
[error]   x POST a chunked body (58 ms)
[error]    an exception was thrown java.lang.IllegalStateException: unexpected message type: DefaultHttpRequest (ClientRouteTestBattery.scala:79)
[error] org.http4s.client.ClientRouteTestBattery$$anonfun$3$$anonfun$apply$19.apply(ClientRouteTestBattery.scala:79)
[error] org.http4s.client.ClientRouteTestBattery$$anonfun$3$$anonfun$apply$19.apply(ClientRouteTestBattery.scala:73)
```

Example from [withFragment PR](https://travis-ci.org/http4s/http4s/jobs/256053023#L1737)
Example from [Multipart PR](https://travis-ci.org/http4s/http4s/jobs/256014078#L2364)

**Comment by rossabaker** [comment]
I probably broke this last night when I merged and went to the beach. 🏖

**Comment by ChristopherDavenport** [comment]
Closed by #1275

**Comment by ChristopherDavenport** [comment]
Finally closed by #1277
